### PR TITLE
Locking expiration date proposal

### PR DESCRIPTION
**Pull request by LucasBrazi06**

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
@OliveGerste Check this PR.
I tested it successfully.

Here is some code I used:
```
let lock = LockingManager.createExclusiveLock(Constants.DEFAULT_TENANT, LockEntity.ASSET, '123456', 1);
console.log('Acquire Lock should be OK');
console.log(await LockingManager.acquire(lock));
await Utils.sleep(2000);

console.log('Acquire Lock should be OK with expiration date');
lock = LockingManager.createExclusiveLock(Constants.DEFAULT_TENANT, LockEntity.ASSET, '123456', 1);
console.log(await LockingManager.acquire(lock));

console.log('Acquire Lock should be KO');
lock = LockingManager.createExclusiveLock(Constants.DEFAULT_TENANT, LockEntity.ASSET, '123456', 1);
console.log(await LockingManager.acquire(lock));
await Utils.sleep(2000);

console.log('Acquire Lock should be OK with expiration date');
lock = LockingManager.createExclusiveLock(Constants.DEFAULT_TENANT, LockEntity.ASSET, '123456', 1);
console.log(await LockingManager.acquire(lock));

```

File: src/locking/LockingManager.ts [hunk 1]
```diff
@@ -17,34 +16,15 @@ const MODULE_NAME = 'LockingManager';
...
  */
 export default class LockingManager {
-  public static createExclusiveLock(tenantID: string, entity: LockEntity, key: string, expirationTimeSeconds?: number): Lock {
-    return this.createLock(tenantID, entity, key, LockType.EXCLUSIVE, expirationTimeSeconds);
+  public static createExclusiveLock(tenantID: string, entity: LockEntity, key: string, lockValiditySecs?: number): Lock {
+    return this.createLock(tenantID, entity, key, LockType.EXCLUSIVE, lockValiditySecs);
   }
 
-  public static async acquire(lock: Lock, timeout?: number): Promise<boolean> {
+  public static async acquire(lock: Lock, timeoutMs?: number, retry = true): Promise<boolean> {
```
Comment (jerome-benoit) [inline src/locking/LockingManager.ts:23]: Putting timeoutMs default to `= 0` will make the API cleaner and the code less complex to understand.

File: src/locking/LockingManager.ts [hunk 2]
```diff
@@ -171,10 +127,64 @@ export default class LockingManager {
       hostname: Utils.getHostname()
     };
     // Set expiration date
-    if (expirationTimeSeconds > 0) {
-      lock.expirationDate = moment(lock.timestamp).add(expirationTimeSeconds, 'seconds').toDate();
+    if (lockValiditySecs > 0) {
+      lock.expirationDate = new Date(lock.timestamp.getTime() + (lockValiditySecs * 1000));
```
Comment (jerome-benoit) [inline src/locking/LockingManager.ts:131]: If you rely on unix timestamp everywhere, you will simplify and speed all expire test and avoid to manipulate `Date` object.

File: src/locking/LockingManager.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/locking/LockingManager.ts:23]: Ok

File: src/locking/LockingManager.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/locking/LockingManager.ts:131]: I'm ok with the date, it's more readable in the DB

File: src/storage/mongodb/LockingStorage.ts [hunk 3]
```diff
@@ -21,15 +22,35 @@ export default class LockingStorage {
...
         $match: filters
       });
     }
+    // Limit records?
+    if (!dbParams.onlyRecordCount) {
+      // Always limit the nbr of record to avoid perfs issues
+      aggregation.push({ $limit: Constants.DB_RECORD_COUNT_CEIL });
+    }
+    // Count Records
+    const logsCountMDB = await global.database.getCollection<any>(Constants.DEFAULT_TENANT, 'locks')
```
Comment (OliveGerste) [inline src/storage/mongodb/LockingStorage.ts:40]: Small spelling issue: 'logs'.

**Review by OliveGerste — APPROVED** [review]
Thanks for the improvement of the original PR.
I wrote some unit tests for expiration and timeout and they still work. So it should be fine. 
I just found a small spelling issue. I can fix it if you want and merge the PR into the original one.

**Comment by jerome-benoit** [comment]
> I just found a small spelling issue. I can fix it if you want and merge the PR into the original one.

Since that PR is on top of your work, you manage the merge.

**Pull request merged**